### PR TITLE
Check Resource Name added to swagger

### PR DESCRIPTION
**Pull request by shreyanshn**
Microsoft.Resources/checkResourceName/action is not defined in swagger #8780

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by raych1 — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 1]
```diff
@@ -518,6 +567,23 @@
...
+          "description": "Name of Resource",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of Resource",
+          "type": "string"
+        },
+        "status": {
+          "description": "Is the resource name allowed or not",
+          "type": "string"
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:582]: Can/should this be an enum?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 2]
```diff
@@ -518,6 +567,23 @@
...
+        "name": {
+          "description": "Name of Resource",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of Resource",
+          "type": "string"
+        },
+        "status": {
+          "description": "Is the resource name allowed or not",
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:581]: So this is a bool-like string?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 3]
```diff
@@ -182,6 +182,55 @@
...
+          "Tenants"
+        ],
+        "operationId": "checkResourceName",
+        "description": "Checks if the resource name is valid or not",
+        "summary": "Checks resource name validity",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:198]: Please break this out into a `$ref`

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 4]
```diff
@@ -182,6 +182,55 @@
...
         }
       }
+    },
+    "/providers/Microsoft.Resources/checkResourceName": {
+      "post": {
+        "tags": [
+          "Tenants"
+        ],
+        "operationId": "checkResourceName",
+        "description": "Checks if the resource name is valid or not",
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:192]: Please explain in a bit more detail what is meant by 'valid' in this context

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 5]
```diff
@@ -182,6 +182,55 @@
...
+            "in": "body",
+            "name": "",
+            "schema": {
+              "required": [
+                "name",
+                "type"
+              ],
+              "properties": {
+                "name": {
+                  "type": "string"
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:208]: Missing description

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 6]
```diff
@@ -182,6 +182,55 @@
...
+                },
+                "type": {
+                  "type": "string"
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:211]: Missing description

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 7]
```diff
@@ -182,6 +182,55 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/providers/Microsoft.Resources/checkResourceName": {
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:186]: Should we go back and update all existing `subscriptions.json` files for older API versions, or is this the only API version that supports this endpoint?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json [hunk 8]
```diff
@@ -518,6 +567,23 @@
...
+        "type": {
+          "description": "Type of Resource",
+          "type": "string"
+        },
+        "status": {
+          "description": "Is the resource name allowed or not",
+          "type": "string"
+        }
+      },
+      "description": "Validity of Resource name for the given type"
```
Comment (anthony-c-martin) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2020-01-01/subscriptions.json:585]: Note - might be worth explaining what this *doesn't* do. For example, `Microsoft.Storage` do their own validation of names and have stricter requirements as they must be globally unique. I wouldn't want customers to think they can use this API to perform the same check.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2015-11-01/subscriptions.json [hunk 9]
```diff
@@ -145,6 +145,40 @@
...
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/CheckResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2015-11-01/subscriptions.json:168]: Missing "default" response

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json [hunk 10]
```diff
@@ -162,6 +162,40 @@
...
+        "parameters": [
+          {
+            "$ref": "#/parameters/CheckResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (KrisBash) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json:186]: Missing default response

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please add "default" response to all new APIs

**Comment by raych1** [comment]
@shreyanshn , please fix the LintDiff check failures.

File: specification/resources/resource-manager/readme.md [hunk 11]
```diff
@@ -616,6 +616,12 @@ input-file:
 - Microsoft.Authorization/stable/2016-12-01/policyAssignments.json
 - Microsoft.Resources/stable/2016-06-01/subscriptions.json
 - Microsoft.Resources/stable/2018-05-01/resources.json
+
+override-info:
+  title: PolicyClient
```
Comment (raych1) [inline specification/resources/resource-manager/readme.md:621]: It should have one value for `title`. Can you refer to other tag sections?

**Review by raych1 — CHANGES_REQUESTED** [review]
The incorrect value in readme.md caused the SDK task failure.

**Comment by raych1** [comment]
@jianyexi , can you help with the lintdiff failures?

**Comment by jianyexi** [comment]
> @jianyexi , can you help with the lintdiff failures?
 
1  override the title in readme, since there are multiple input swaggers 
2   For the following error message output by the check of commit: `single override`, the   `$.definitions.Operation.description` has conflict  among input swaggers  , the value of `$.definitions.Operation.description` should be the same in every swaggers
```
$.definitions.Operation.description' has incompatible values (---\n" +
  'Microsoft.Authorization operation\n' +
  ', ---\n' +
  'Microsoft.Resources operation\n' +
  ').\n' 
```

**Comment by raych1** [comment]
@shreyanshn , did you figure out how to fix the lint failures?

**Comment by shreyanshn** [comment]
@raych1 I cannot fix the incompatible values for the operation part of the swagger because it is not a part of the API that I changed. Not sure how to fix that.

**Comment by jianyexi** [comment]
@raych1  I have ran lint locally and no new lint error reported , so I think it can be force merged

```
SDK Errors/Warnings
===================
Errors:    Before:  18  - After:  18
Warnings:  Before:  9  - After:  9
New SDK Errors:  0
New SDK Warnings:  0
Existing SDK Errors:  0
Existing SDK Warnings:  0
ARM Errors/Warnings
===================
Errors:    Before:  144  - After:  144
Warnings:  Before:  18  - After:  18
New ARM Errors:  0
New ARM Warnings:  0
Existing ARM Errors:  0
Existing ARM Warnings:  0
-----------------------------------------
```

**Comment by raych1** [comment]
> @raych1 I have ran lint locally and no new lint error reported , so I think it can be force merged
> 
> ```
> SDK Errors/Warnings
> ===================
> Errors:    Before:  18  - After:  18
> Warnings:  Before:  9  - After:  9
> New SDK Errors:  0
> New SDK Warnings:  0
> Existing SDK Errors:  0
> Existing SDK Warnings:  0
> ARM Errors/Warnings
> ===================
> Errors:    Before:  144  - After:  144
> Warnings:  Before:  18  - After:  18
> New ARM Errors:  0
> New ARM Warnings:  0
> Existing ARM Errors:  0
> Existing ARM Warnings:  0
> -----------------------------------------
> ```

Thanks @jianyexi !

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
@akning-ms , as synced with you offline, lintDiff fails by autorest error on different description value for same property which is legacy file. Jianye ran the lintDiff offline and the result looks good for the change. Please help force merge this PR.

**Comment by akning-ms** [comment]
@nullMDR @changlong-liu for awareness track 2 python sdk generation fail for this PR

**Pull request merged**